packages/app-desktop/gui/NoteEditor/utils/useFormNote.ts

```diff
@@ -223,6 +207,25 @@ export default function useFormNote(dependencies: HookDependencies) {
 		// eslint-disable-next-line @seiyab/react-hooks/exhaustive-deps -- Old code before rule was applied
 	}, [noteId, isProvisional, formNote]);
 
+	useEffect(() => {
+		if (formNote.id && formNote.id !== previousNoteId && isProvisional) {
+			const focusSettingName = formNote.is_todo ? 'newTodoFocus' : 'newNoteFocus';
+
+			const autoFocusInterval = setInterval(() => {
+				if (editorRef.current && titleInputRef.current) {
+					if (Setting.value(focusSettingName) === 'title') {
+						titleInputRef.current.focus();
+						clearInterval(autoFocusInterval);
+					} else {
+						editorRef.current.execCommand({ name: 'editor.focus' });
+						clearInterval(autoFocusInterval);
+					}
+				}
+			}, 2);
+		}
+		// eslint-disable-next-line @seiyab/react-hooks/exhaustive-deps -- Old code before rule was applied
+	}, [formNote.id, previousNoteId, isProvisional]);
+
```

Comment:
Just a few things:

- Please remove the comment "eslint-disable-next-line" - new code must pass the linter
- You are not clearing the interval when the effect is unmounted - you need to do that in the cleanup handler. See https://react.dev/reference/react/useEffect and search for "cleanup"

Reply:
Wouldn't it be enough to check only "isProvisional"? Maybe you don't need formNote.id and previousNoteId?

Reply:
My bad, cleanup function slipped my mind. I've applied the rest of the changes and you're right that we only needed to check isProvisional, but we still run into the same issue as before. Debugger shows everything working as intended; however in regular usage, focusing on the title works but not on the editor again and editorRef in CodeMirror still shows as null.

Reply:
Ok so I'll need to look into it and debug it myself then